### PR TITLE
ddl: Fix the data race in foreign_key_test.go

### PR DESCRIPTION
**Pull request by zimulala**
Add lock to fix data race

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: ddl/foreign_key_test.go [hunk 1]
```diff
@@ -159,22 +160,28 @@ func (s *testForeighKeySuite) TestForeignKey(c *C) {
...
+	// fix data race
+	var mu sync.Mutex
 	checkOK = false
 	tc.onJobUpdated = func(job *model.Job) {
 		if job.State != model.JobDone {
 			return
 		}
 		t := testGetTable(c, d, s.dbInfo.ID, tblInfo.ID)
 		s.testForeignKeyExist(c, t, "c1_fk", false)
+		mu.Lock()
```
Comment (hanfei1991) [inline ddl/foreign_key_test.go]: Need we all lock for line 147?

**Comment by zimulala** [comment]
PTAL @hanfei1991

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**